### PR TITLE
Fix clrcompression testhost path

### PR DESCRIPTION
**Pull request by joperezr**
cc: @AlexGhiondea @weshaggard 

When submitting #17509 PR I didn't add configuration to the path where we are dumping clrcompression, so I'm fixing that with this PR.

### CODE REVIEW

**Comment by weshaggard** [comment]
See my comment in the other PR where I suggest we should move the bin-placing of the native bits into a step at the end of our managed build. That way we can avoid all the duplication.

**Comment by joperezr** [comment]
Done. Now the Native-BinPlace.proj will take care of binplacing the native assets (if any) into the appropiate dirs so that we avoid duplication of the logic.

File: src/Native/Native-BinPlace.proj [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <Target Name="Clean" />
+  <Target Name="Rebuild" DependsOnTargets="Clean;Build" />
+
+  <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
+
+  <Target Name="GetBinPlaceItems">
+    <ItemGroup>
+      <BinPlaceItem Include="$(NativeBinDir)/*.dll" />
+      <BinPlaceItem Include="$(NativeBinDir)/*.pdb" />
+      <BinPlaceItem Include="$(NativeBinDir)/*.lib" />
```
Comment (weshaggard) [inline src/Native/Native-BinPlace.proj]: do we really need .lib and .a files?

Comment (weshaggard) [inline src/Native/Native-BinPlace.proj]: If this ends up being all the file types we might as well just do *.

File: src/dirs.proj [hunk 2]
```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <Project Include="ref.builds" />
     <Project Include="shims\shims.proj" />
+    <Project Include="Native\Native-BinPlace.proj" />
```
Comment (weshaggard) [inline src/dirs.proj]: Can you please make this file lower case?

**Comment by weshaggard** [comment]
FYI: @mellinoe

File: src/Native/Native-BinPlace.proj (hunk 1 above)
Comment (mellinoe) [inline src/Native/Native-BinPlace.proj]: I don't think we need those in the test runtime, plus the `*.a` files.

**Comment by mellinoe** [comment]
The solution works and will be future-proof if we move paths around in the managed build, but I'm worried there's going to be a lot of confusion when people try to re-build their native stuff and then re-run their tests (without knowing they also need to run this). Is it feasible to have the native build scripts call `msbuild native-binplace.proj`, or at least print out that it needs to be done?

File: src/dirs.proj (hunk 2 above)
Comment (joperezr) [inline src/dirs.proj]: will change.

File: src/Native/Native-BinPlace.proj (hunk 1 above)
Comment (joperezr) [inline src/Native/Native-BinPlace.proj]: I'm not sure but this list contains all of the file extensions of what we were copying down to the runtime/testhost dirs before, which is how I came up with it. `.lib` is only on Window's clrcompression and `.a` is in non-Windows' System.Natve.a

**Comment by joperezr** [comment]
> Is it feasible to have the native build scripts call `msbuild native-binplace.proj`, or at least print out that it needs to be done?

I believe it should be ok to call the project at the end. I'll take a stab at it.

**Comment by mellinoe** [comment]
> I believe it should be ok to call the project at the end. I'll take a stab at it.

On second thought, we need to preserve the ability to do native builds without any managed code running. It should be conditional in some way (not sure how to detect that, exactly. Perhaps just check if the build tools have been restored?

**Comment by joperezr** [comment]
@dotnet-bot test this please

**Comment by joperezr** [comment]
@dotnet-bot test Innerloop Tizen armel Debug Cross Build please
@dotnet-bot test Innerloop Tizen armel Release Cross Build please

**Comment by joperezr** [comment]
I'm not sure why the Tizen builds are failing, I don't think my changes should have impacted that

**Comment by joperezr** [comment]
cc: @hseok-oh I'm not sure why these changes are causing the Tizen build to fail and by just looking at the logs it's not obvious what it is. What I'm doing with this PR is to change the place where we have the logic of copying out our native assets. Given that you added these builds, do you mind taking a look?

**Comment by hseok-oh** [comment]
@joperezr 
https://github.com/dotnet/corefx/blob/master/cross/arm32_ci_script.sh#L247-L265
`__runtimePath` directory in this script was made by build-native.sh, but by this commit, this directory is not generated. And I heard that we cannot armel managed build yet, so that path is not generated by build-managed.sh.

Here is simple temporary fix
https://github.com/hseok-oh/corefx/tree/ci/armel_ci_fix

BTW, arm32_ci_script.sh is too messy. I'll make a commit to clean up it, soon.

**Pull request merged**